src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> {
+            if (!cache.containsKey(userId)) {
+                cache.put(userId, new UserData(userId, "Details for user" + userId));
+            }
+            return cache.get(userId);
```

Comment:
You should add a simulated task, like fetching user data from a database (mock this with a delay).